contracts/test/StakeTableV2Commission.t.sol

```diff
@@ -0,0 +1,209 @@
+
+        // Pause the contract
+        vm.prank(pauser);
+        proxy.pause();
+
+        // Try to update commission while paused
+        vm.startPrank(validator);
+        vm.expectRevert(PausableUpgradeable.EnforcedPause.selector);
+        proxy.updateCommission(1000);
+        vm.stopPrank();
+    }
+
+    function test_CommissionUpdate_DecreaseMaxDelta() public {
+        address validator = makeAddr("validator");
+        uint16 maxCommission = 10000;
+        stakeTableUpgradeTest.registerValidatorOnStakeTableV2(
+            validator, "123", maxCommission, proxy
+        );
+
+        vm.startPrank(validator);
+
+        vm.expectEmit(true, false, false, true);
+        uint16 minCommission = 0;
+        emit StakeTableV2.CommissionUpdated(validator, block.timestamp, minCommission);
+        proxy.updateCommission(minCommission);
+        vm.stopPrank();
+    }
+
+    function test_SetMinCommissionUpdateInterval_Success() public {
+        uint256 newInterval = 14 days;
+
+        vm.startPrank(stakeTableUpgradeTest.admin());
+        vm.expectEmit(true, false, false, true);
+        emit StakeTableV2.MinCommissionUpdateIntervalUpdated(newInterval);
+        proxy.setMinCommissionUpdateInterval(newInterval);
+
+        assertEq(proxy.minCommissionUpdateInterval(), newInterval);
+        vm.stopPrank();
+    }
+
+    function test_SetMaxCommissionIncrease_Success() public {
+        uint16 newMaxIncrease = 1000; // 10%
+
+        vm.startPrank(stakeTableUpgradeTest.admin());
+        vm.expectEmit(true, false, false, true);
+        emit StakeTableV2.MaxCommissionIncreaseUpdated(newMaxIncrease);
+        proxy.setMaxCommissionIncrease(newMaxIncrease);
+
+        assertEq(proxy.maxCommissionIncrease(), newMaxIncrease);
+        vm.stopPrank();
+    }
+
+    function test_SetMinCommissionUpdateInterval_RevertWhenNotOwner() public {
+        address notAdmin = makeAddr("notAdmin");
+        uint256 newInterval = 14 days;
+
+        vm.startPrank(notAdmin);
+        vm.expectRevert(
+            abi.encodeWithSelector(OwnableUpgradeable.OwnableUnauthorizedAccount.selector, notAdmin)
+        );
+        proxy.setMinCommissionUpdateInterval(newInterval);
+        vm.stopPrank();
+    }
+
+    function test_SetMaxCommissionIncrease_RevertWhenNotOwner() public {
+        address notAdmin = makeAddr("notAdmin");
+        uint16 newMaxIncrease = 1000;
+
+        vm.startPrank(notAdmin);
+        vm.expectRevert(
+            abi.encodeWithSelector(OwnableUpgradeable.OwnableUnauthorizedAccount.selector, notAdmin)
+        );
+        proxy.setMaxCommissionIncrease(newMaxIncrease);
+        vm.stopPrank();
+    }
+
+    function test_DefaultValues() public view {
+        // This is the only test that checks default values - if defaults change, only this test
+        // should fail
+        assertEq(proxy.minCommissionUpdateInterval(), 7 days);
+        assertEq(proxy.maxCommissionIncrease(), 500);
+    }
+}
```

Comment:
we should add a test for handling an upgrade from v1 to v2 where the initial commission array is not empty

Reply:
There are rust tests for this, but sure can also add a solidity test.